php/Terminus/Collections/Hostnames.php

```diff
@@ -58,6 +53,20 @@ public function create($hostname)
     public function setHydration($value)
     {
         $this->hydrate = $value;
+        $this->setURL();
         return $this;
     }
+
+    /**
+     * Sets the URL to fetch this collection
+     */
+    protected function setURL()
```

Comment:
I generally favor using a getter for something like this so you don't have to worry about maintaining the state when things change like this, but since this is modifying the old collection it probably makes sense to leave this as is.